### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by cspray**
- Updates the README to be formatted more generally in-line with the
primary amphp/amp package.

- Updates the examples the README to properly return a Promise and use
appropriate method names.

---

I also would like to have a clarifying question around the documentation on `Websocket::onHandshake`, specifically the `@return` documentation:

```
@return Promise<Response> Resolve with the given response to accept the connection or resolve with a new Response object to deny the connection.
```

This phrasing implies that  a brand new Response instance needs to be created to deny the connection. However, the example in the README, and I believe the more intuitive API, is for any non-2XX response to deny the connection. I am unsure as to the exact semantics here but I would like to either clarify the Websocket docblocks or the example to be consistent.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,4 +3,5 @@ build
 composer.lock
 phpunit.xml
 vendor
-reports
\ No newline at end of file
+reports
+.idea
```
Comment (bwoebi) [inline .gitignore]: sigh, why do people always want to add it?
`.idea` shall be part of your own global `.gitignore`; not our responsibility.

**Comment by cspray** [comment]
I understand that the .gitignore was bothersome and I removed it. However, the bigger piece about my question between the discrepancy in the documentation and the example still needs an answer.

**Comment by trowski** [comment]
A new instance does not have to be returned. The docs should say that the `Response` instance can be modified to be a non-101 response to effectively deny the connection, though generally a 4xx response would be appropriate. The docs should make it clear that the response object can also have headers added, such as cookies, if desired.

File: README.md [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
+> Currently this library is undergoing a RC phase on a push to 2.0! Please check out the 2.0 RC releases
+
+## Documentation
+
+The documentation for this library is currently a work in progress. Pull Requests to improve the documentation 
+are always welcome!
+
+## Requirements
+
+- PHP 7.0+
```
Comment (trowski) [inline README.md]: Should be 7.1.

**Comment by cspray** [comment]
@trowski @bwoebi Got the tests updated from master to get those passing and addressed all of the feedback. Believe this is good to be merged in. Please let me know if there's anything else I can add.

File: README.md [hunk 3]
```diff
@@ -66,3 +92,7 @@ Loop::run(function () use ($server) {
     yield $server->start();
 });
 ```
+
+[amphp/http-server]: https://github.com/amphp/http-server
+[Composer]: https://getcomposer.org
+[Request Handler]: https://amphp.org/http-server/classes/request-handler
```
Comment (kelunik) [inline README.md]: I prefer these to be inline.

**Review by kelunik — APPROVED** [review]

**Comment by kelunik** [comment]
Thanks!

**Pull request merged**